packcheck.sh

```diff
@@ -1295,10 +1295,33 @@ create_and_unpack_pkg_dist() {
         | sort | grep -v '^$' > .packcheck/tar-ztf1.txt
     fi
     git ls-files | sort | grep -v '^$' > .packcheck/git-ls-files.txt
-    diff -B --suppress-common-lines .packcheck/tar-ztf1.txt .packcheck/git-ls-files.txt ||
+    local diff_res_file=".packcheck/tar-git-diff.txt"
+    local tar_minus_git
+    local git_minus_tar
+    diff -B --suppress-common-lines .packcheck/tar-ztf1.txt .packcheck/git-ls-files.txt > "$diff_res_file" ||
       { echo "WARNING! Source distribution tar and git repo contents differ."
+        tar_minus_git="$(awk '$0~/^< .+$/' "$diff_res_file")"
+        git_minus_tar="$(awk '$0~/^> .+$/' "$diff_res_file")"
+        if test -n "$tar_minus_git"
+        then
+            echo
+            echo "The following files exist in your source distribution but \
+are not committed to the version control."
+            echo "$tar_minus_git"
+            echo "Please consider committing them to your version control."
+        fi
+        if test -n "$git_minus_tar"
+        then
+            echo
+            echo "The following files are committed to the version control \
+but do not exist in the source distribution."
+            echo "$git_minus_tar"
+            echo "Please consider adding them to your cabal file under \
+'extra-source-files' or 'extra-doc-files'."
```

Comment:
Or if you do not want to add them in the source distribution then add them to `.packcheck.ignore` at the root of the git repository.